test/fetchEnvs-tests.js

```diff
@@ -0,0 +1,382 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const assert = require('chai').assert;
+const FetchEnvs = require('../lib/FetchEnvs');
+const fs = require('fs-extra');
+
+
+const krm = {
+  _apiVersion: 'deploy.razee.io/v1alpha2',
+  _kind: 'MustacheTemplate',
+  async request(reqOpt) {
+    const ref = {
+      apiVersion: reqOpt.uri.apiVersion,
+      kind: reqOpt.uri.kind,
+      name: reqOpt.uri.name,
+      namespace: reqOpt.uri.namespace,
+      labelSelector: reqOpt?.qs?.labelSelector
+    };
+
+    const res = await kubeGetResource(ref);
+    return res;
+  },
+  async get(name, namespace) {
+    const ref = this.uri({ name, namespace });
+
+    return await kubeGetResource(ref);
+  },
+  uri(options) {
+    return { ...options, apiVersion: this._apiVersion, kind: this._kind, };
+  }
+};
+
+const controllerObject = {
+  data: {
+    type: 'ADDED',
+    object: {
+      apiVersion: 'deploy.razee.io/v1alpha2',
+      kind: 'MustacheTemplate',
+      metadata: {
+        name: 'rd-test',
+        namespace: 'razeedeploy',
+      },
+      spec: {
+        clusterAuth: { impersonateUser: 'razeedeploy' },
+        templateEngine: 'handlebars',
+        envFrom: [],
+        env: [],
+        tempates: [],
+        strTemplates: [],
+      }
+    }
+  },
+  kubeResourceMeta: krm,
+  kubeClass: {
+    getKubeResourceMeta: (apiVersion, kind) => {
+      const newKRM = { ...krm };
+      newKRM._apiVersion = apiVersion;
+      newKRM._kind = kind;
+      return newKRM;
+    }
+  },
+};
+
+async function kubeGetResource(ref) {
+  const kubeData = await fs.readJSON(`${__dirname}/fetchEnvs-test-scenarios/sampleData.json`);
+  const {
+    name,
+    labelSelector,
+    namespace,
+    kind,
+    apiVersion
+  } = ref;
+  if (!kubeData[kind]) {
+    return;
+  }
+
+  let fn = labelSelector ? 'filter' : 'find';
+  let lookup = kubeData[kind][fn](obj => {
+    let match = true;
+    match = (obj.apiVersion === apiVersion && match) ? true : false;
+    match = (obj.kind === kind && match) ? true : false;
+    match = ((obj.metadata.name === name || labelSelector !== undefined) && match) ? true : false;
+    match = (obj.metadata.namespace === namespace && match) ? true : false;
+    if (labelSelector) {
+      const objLabels = obj.metadata.labels ?? {};
+      labelSelector.split(',').forEach(label => {
+        let [key, value] = label.split('=');
+        match = (objLabels[key] === value && match) ? true : false;
+      });
+    }
+    return match;
+  });
+  return labelSelector ? { items: lookup } : lookup;
+}
+
+describe('fetchEnvs', function () {
+  afterEach(function () {
+    controllerObject.data.object.spec.envFrom.splice(0, controllerObject.data.object.spec.envFrom.length);
+    controllerObject.data.object.spec.env.splice(0, controllerObject.data.object.spec.env.length);
```

Comment:
might be a more obvious to just assign a new array
```javascript
controllerObject.data.object.spec.envFrom = []
```

Reply:
was reusing the array to save garbage collector some work

Reply:
guess its not that big a deal in testing

Reply:
updated